pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/series/TimeSeriesBlock.java

```diff
@@ -33,11 +34,12 @@ public class TimeSeriesBlock {
   private final TimeBuckets _timeBuckets;
   private final Map<Long, List<TimeSeries>> _seriesMap;
 
-  public TimeSeriesBlock(TimeBuckets timeBuckets, Map<Long, List<TimeSeries>> seriesMap) {
+  public TimeSeriesBlock(@Nullable TimeBuckets timeBuckets, Map<Long, List<TimeSeries>> seriesMap) {
```

Comment:
Why timebucket is nullable? What will happen if TimeBucket is null? Is this for Instant query use case?

Reply:
Yeah this is for that, and also for the case when we have to perform partial aggregates. in that case, we might not be able to bucket the values at a time granularity under the combine operator, and may have to instead return time values as a Long[] instead of TimeBuckets in the TimeSeries.